gcloud-java-storage/src/main/java/com/google/gcloud/storage/StorageImpl.java

```diff
@@ -668,7 +668,7 @@ public T apply(BatchResponse.Result<T> result) {
       checkArgument(prev == null, "Duplicate option %s", option);
     }
     Boolean value = (Boolean) temp.remove(DELIMITER);
-    if (Boolean.TRUE.equals(value)) {
+    if (Boolean.FALSE.equals(value)) {
```

Comment:
Do you think we should also set it if value is `null`? Basically, what should be the default if recursive option was not specified? Whatever we pick, I think it would be worth to document
the default. Also, if we pick "non-recursive" when not specifying than we can consider removing the boolean param from the option.